go/pkg/apis/enricher/framework/javascript/nodejs/express_detector.go

```diff
@@ -31,8 +32,8 @@ func (e ExpressDetector) DoFrameworkDetection(language *model.Language, config s
 	}
 }
 
-func (e ExpressDetector) DoPortsDetection(component *model.Component) {
-	files, err := utils.GetFilePathsFromRoot(component.Path)
+func (e ExpressDetector) DoPortsDetection(component *model.Component, ctx *context.Context) {
+	files, err := utils.GetCachedFilePathsFromRoot(component.Path, ctx)
```

Comment:
Unfortunately I cannot extend the context class as it's defined in a different package. The only thing i can do is to create a custom sub-package in my code. Something like
```
type myContext context.Context 
or
type myContext struct {
   *context.Context
}

then
func (c myContext) GetCachedFilePathsFromRoot
```
So it will become 
```
func (e ExpressDetector) DoPortsDetection(component *model.Component, ctx *myContext) {
	files, err := ctx.GetCachedFilePathsFromRoot(component.Path)
```
Do you prefer having this?

Reply:
OK I though context was an alyzer thing I did not catch it came from Go !!!!